### PR TITLE
CommandHandler is not an entity

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
This PR:
* Extracts all code shared by objects that handle commands as static methods in `CommandHandlingMethod`.
* Makes `CommandHandler` an interface with abstract implementation in `AbstractCommandHandler`, which is no longer an entity.
* Isolates code required for working with handling methods under the `reflect` package.

Some more work is expected in following PR(s) for hiding event-related reflection-based code.

### CODE REVIEW

**Comment by alexander-yevsyukov** [comment]
@armiol PTAL

File: server/src/main/java/org/spine3/server/aggregate/AggregateRoot.java [hunk 1]
```diff
@@ -63,6 +65,34 @@ protected AggregateRoot(BoundedContext boundedContext, I id) {
     }
 
     /**
+     * Creates a new {@code AggregateRoot} entity and sets it to the default state.
+     *
+     * @param id        the ID of the {@code AggregatePart} is managed by {@code AggregateRoot}
```
Comment (armiol) [inline server/src/main/java/org/spine3/server/aggregate/AggregateRoot.java]: `is` seems unnecessary.

File: server/src/main/java/org/spine3/server/reflect/CommandHandlerMethod.java [hunk 2]
```diff
@@ -67,11 +116,99 @@ public CommandHandlerMethod(Method method) {
...
+
+    /**
+     * Creates new {@code CommandContext} with passed parameters.
+     *
+     * @param producerId the ID of an object which produced the event
+     * @param version optional version of the object
+     * @param commandContext the context of the command handling of which produced the event
+     * @return new {@code CommandContext}
+     */
+    public static EventContext createEventContext(Any producerId,
```
Comment (armiol) [inline server/src/main/java/org/spine3/server/reflect/CommandHandlerMethod.java]: As long as it is `public`, we should ensure both `producerId` and `commandContext` aren't `null`s.

I also assume we don't have a nullability check for the `static` methods of this class in tests.

File: server/src/main/java/org/spine3/server/reflect/CommandHandlerMethod.java [hunk 3]
```diff
@@ -67,11 +116,99 @@ public CommandHandlerMethod(Method method) {
...
+                                                         .setTimestamp(timestamp)
+                                                         .setCommandContext(commandContext)
+                                                         .setProducerId(producerId);
+        if (version != null) {
+            builder.setVersion(version);
+        }
+        return builder.build();
+    }
+
+    public static List<Event> toEvents(final Any producerId,
```
Comment (armiol) [inline server/src/main/java/org/spine3/server/reflect/CommandHandlerMethod.java]: Same as above.

File: server/src/main/java/org/spine3/server/reflect/EventApplierMethod.java [hunk 4]
```diff
@@ -35,20 +36,48 @@
...
-    EventApplierMethod(Method method) {
+    private EventApplierMethod(Method method) {
         super(method);
     }
 
+    static EventApplierMethod from(Method method) {
+        return new EventApplierMethod(method);
+    }
+
+    public static EventApplierMethod forEventMessage(Class<? extends Aggregate> cls,
```
Comment (armiol) [inline server/src/main/java/org/spine3/server/reflect/EventApplierMethod.java]: Let's ensure neither `cls` nor `eventMessage` is `null`.

File: server/src/main/java/org/spine3/server/reflect/EventSubscriberMethod.java [hunk 5]
```diff
@@ -36,34 +41,77 @@
...
         super(method);
     }
 
     /**
+     * Obtains the method for handling the event in the passed class.
+     *
+     * @throws IllegalStateException if the passed class does not have an event handling method
+     *                               for the class of the passed message
+     */
+    public static EventSubscriberMethod forMessage(Class<?> cls, Message eventMessage) {
```
Comment (armiol) [inline server/src/main/java/org/spine3/server/reflect/EventSubscriberMethod.java]: Let's ensure the parameters aren't `null`s.

**Comment by alexander-yevsyukov** [comment]
@armiol PTAL

**Pull request merged**